alf/environments/igibson/suite_igibson.py

```diff
@@ -0,0 +1,384 @@
+            print(e)
+
+        rgbd_low = np.vstack((rgb_space.low, depth_space.low))
+        rgbd_high = np.vstack((rgb_space.high, depth_space.high))
+        assert rgbd_low.shape == rgbd_high.shape
+        rgbd_space = gym.spaces.Box(
+            low=rgbd_low, high=rgbd_high, dtype=np.uint8)
+
+        observation_space.spaces.pop('rgb')
+        observation_space.spaces.pop('depth')
+        observation_space.spaces['rgbd'] = rgbd_space
+        return observation_space
+
+    def observation(self, observation):
+        return self.transform_observation(observation)
+
+    def transform_observation(self, observation):
+        rgbd_obs = np.vstack((observation['rgb'], observation['depth']))
+        observation.pop('rgb')
+        observation.pop('depth')
+        observation['rgbd'] = rgbd_obs
+        return observation
+
+
+@alf.configurable()
+class OneHotIntWrapper(BaseObservationWrapper):
+    """Transforms the data type of one-hot input from float to int"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        obs_low = np.full_like(
+            observation_space.low, fill_value=0, dtype=np.uint8)
+        obs_high = np.full_like(
+            observation_space.high, fill_value=255, dtype=np.uint8)
+        assert obs_low.shape == obs_high.shape
+        observation_space = gym.spaces.Box(
+            low=obs_low, high=obs_high, dtype=np.uint8)
+        return observation_space
+
+    def transform_observation(self, observation):
+        return (observation * 255).astype(np.uint8)
+
+
+@alf.configurable()
+class ImageIntWrapper(BaseObservationWrapper):
+    """Transforms the data type of image input from float to int"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        obs_low = np.full_like(
+            observation_space.low, fill_value=0, dtype=np.uint8)
+        obs_high = np.full_like(
+            observation_space.high, fill_value=255, dtype=np.uint8)
+        assert obs_low.shape == obs_high.shape
+        observation_space = gym.spaces.Box(
+            low=obs_low, high=obs_high, dtype=np.uint8)
+        return observation_space
+
+    def transform_observation(self, observation):
+        return (observation * 255).astype(np.uint8)
+
+
+@alf.configurable()
+class RenderWrapper(gym.Wrapper):
+    """Implements render() for iGibson env for visualization purposes"""
+
+    def __init__(self, env):
+        super().__init__(env)
+        self.env = env
+        self.metadata['render.modes'] = ['rgb_array', 'human']
+
+    def render(self, mode='human', **kwargs):
+        if 'vision' in self.env.sensors:
+            vision_obs = self.sensors['vision'].get_obs(self)
+            if 'rgb' in vision_obs:
+                rgb_obs = vision_obs['rgb']
+                rgb_obs = (rgb_obs * 255).astype(np.uint8)
```

Comment:
If both 'rgb' and 'depth' are generated as uint8 in the first place by the simulator, then the conversions here can be skipped.